sections/image-with-text.liquid

```diff
@@ -14,11 +14,46 @@
   }
 {%- endstyle -%}
 
-<div class="image-with-text image-with-text--{{ section.settings.content_layout }} page-width isolate{% if settings.text_boxes_border_thickness > 0 and settings.text_boxes_border_opacity > 0 and settings.media_border_thickness > 0 and settings.media_border_opacity > 0 %} collapse-borders{% endif %}{% unless section.settings.color_scheme == 'background-1' and settings.media_border_thickness > 0 and settings.text_boxes_shadow_opacity == 0 and settings.text_boxes_border_thickness == 0 or settings.text_boxes_border_opacity == 0 %} collapse-corners{% endunless %} section-{{ section.id }}-padding">
+{%- liquid
+  if settings.text_boxes_border_thickness > 0 and settings.text_boxes_border_opacity > 0 and settings.media_border_thickness > 0 and settings.media_border_opacity > 0
+    assign borders_class = ' collapse-borders'
+  endif
+
+  unless section.settings.color_scheme == 'background-1' and settings.media_border_thickness > 0 and settings.text_boxes_shadow_opacity == 0 and settings.text_boxes_border_thickness == 0 or settings.text_boxes_border_opacity == 0
+    assign corners_class = ' collapse-corners'
+  endunless
+
+  if section.settings.image_shape != 'default' and section.settings.color_scheme == 'background-1'
+    assign borders_class = nil
+    assign corners_class = nil
+    assign hide_borders_class = ' hide-borders'
+    assign hide_shadow_class = ' hide-shadow'
+  endif
+
+  if section.settings.image_shape != 'default'
+    assign shape_class = ' image--shape'
+  endif
+
+  unless section.settings.image_shape == 'default' or section.settings.image_shape != 'default' and section.settings.color_scheme == 'background-1' or section.settings.content_layout == 'overlap'
```

Comment:
Nit: This line is confusing to me because it's mixing `or` and `and`. Suggestion:
```liquid
unless section.settings.image_shape == 'default' or section.settings.color_scheme == 'background-1' or section.settings.content_layout == 'overlap'
```

Reply:
_This code will move over to #2374, so we'll address this comment there._